### PR TITLE
Move to using Feedjira 3.0

### PR DESCRIPTION
**Pull request by mikeastock**
Hello 👋! I'm a maintainer on Feedjira and are hoping you guys are open to potentially testing out and giving us feedback on the new interface in Feedjira 3.0.

Feedjira removed it's built-in HTTP fetching so that has been replaced with HTTParty. The underlying parsing code has not changed and Feedjira is in a beta status while we try to nail down the new interface.

We are open to any and all feedback about Feedjira. Thanks!

### CODE REVIEW

**Comment by swanson** [comment]
Thanks, I'm glad to see Feedjira is alive and well. Just to set expectations, this project is fairly dormant at the moment; I'm open to upgrading, but I personally can't commit to testing it out. 

If some folks would like to help run this branch for a few days and if there are no issues, let's merge away.

**Comment by gabrielpoca** [comment]
Hi! I'm going to deploy this change to my personal instance and run it for two weeks. I'll let you know if it's working fine.

**Comment by mikeastock** [comment]
@gabrielpoca fantastic! Let me know if you have any issues at all. Thanks!

**Comment by fakefarm** [comment]
Hello friends 👋,
I pulled down this branch and can confirm FeedJira V3 works as advertised on `Ruby 2.3.3`.
Initially, I tried installing with `Ruby 2.5.0` and `2.4.1` without success. I may dig into that later.

Dave

**Comment by gabrielpoca** [comment]
I think it's not working for me, I have not received a story in almost a week. I'll try to look into it later.

File: app/utils/feed_discovery.rb [hunk 1]
```diff
@@ -1,20 +1,22 @@
...
-      get_feed_for_url(urls.first, parser) do
+      get_feed_for_url(urls.first, parser, client) do
         return false
       end
     end
   end
 
-  def get_feed_for_url(url, parser)
-    feed = parser.fetch_and_parse(url)
+  def get_feed_for_url(url, parser, client)
```
Comment (gabrielpoca) [inline app/utils/feed_discovery.rb]: Hi @mikeastock , I looked into it a little more and it seems that some feeds are no longer working. I tracked the issue to this line. I tried it on racksh. The following doesn't work:

```
FeedDiscovery.new.get_feed_for_url('http://mkremins.github.io/atom.xml', Feedjira, HTTParty)
```

It returns nil. While the previous code was working:

```
Feedjira::Feed.fetch_and_parse('http://mkremins.github.io/atom.xml')
```

Do you know what's happening?

Comment (mikeastock) [reply app/utils/feed_discovery.rb]: I'll dig into this this weekend and get back to you.

Comment (dwaite) [reply app/utils/feed_discovery.rb]: @mikeastock it appears the issue was that HTTParty.get returns a response, not a string body. Adding two `.to_s` fixed things for me

Comment (dperson) [reply app/utils/feed_discovery.rb]: @dwaite I'd like to merge this to my fork at least seeing that nothing seems to be merged on mainline anymore. May have to migrate to another reader eventually. Was the fix you tested out appending the `.to_s` to the following 2 lines? Thanks.

File: app/tasks/fetch_feed.rb [hunk 2]
```diff
@@ -30,7 +32,8 @@ def fetch
   private
 
   def fetch_raw_feed
-    @parser.fetch_and_parse(@feed.url)
+    response = @client.get(@feed.url)
+    @parser.parse(response)
```
Comment (dperson) [inline app/tasks/fetch_feed.rb]: @dwaite these lines follow the same pattern... should they have the `.to_s` appended as well? Assuming that I understood your previous fix (bellow) correctly.

**Comment by mockdeep** [comment]
@mikeastock I know it's been forever since this has been opened, but are you able to follow up on this?

**Comment by mockdeep** [comment]
Closing this, superceded by https://github.com/swanson/stringer/pull/535. Thanks for getting the ball rolling on this @mikeastock!

**Pull request closed without merging**